crates/solver/src/solver/single_order_solver.rs

```diff
@@ -105,6 +105,20 @@ impl From<anyhow::Error> for SettlementError {
     }
 }
 
+// Used by the single order solvers to verify that the response respects the order price.
+// We have also observed that a 0x buy order did not respect the queried buy amount so verifying
+// just the price or verifying just one component of the price (sell amount for buy orders, buy
+// amount for sell orders) is not enough.
+pub fn execution_respects_order(
+    order: &LimitOrder,
+    executed_sell_amount: U256,
+    executed_buy_amount: U256,
+) -> bool {
+    // note: This would be different for partially fillable orders but LimitOrder does currently not
+    // contain the remaining fill amount.
+    executed_sell_amount <= order.sell_amount && executed_buy_amount >= order.buy_amount
+}
```

Comment:
We can copy the tests from `balancer_sor.rs` here:
```rust
    #[test]
    fn test_satisfies_limit_price() {
        // ...
    }
```

Reply:
I'm just wondering if that is useful. This function is very simple so I don't feel a test will catch an error. Either the logic is right or I will make the same mistake in the test too. Don't mind adding it if other people think it is useful.

Reply:
Agree in general that this function is quite simple, but since the was already there, it is useful for verifying that there are no regressions with the refactor.

Reply:
Added tests. One (expected) difference is that the new function allows the input amount to unexpectedly be less as long as the out amount is still respected. The sor solver was more strict than necessary here even though it would be weird for the api to return that.